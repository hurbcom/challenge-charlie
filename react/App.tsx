import React, { Suspense } from "react";
import ErrorBoundary from "./ErrorBoundary";
import { GlobalStyle } from "./styles/global";
import HeroImage from "@components/HeroImage";
import WeatherAccordeon from "@components/WeatherAccordeon";

const App: React.FC = () => {
    return (
        <Suspense fallback={<div>Suspending...</div>}>
            <ErrorBoundary>
                <GlobalStyle />
                <HeroImage
                    image={
                        "https://www.bing.com/th?id=OHR.KitsAspen_PT-BR8299899730_1920x1080.jpg&rf=LaDigue_1920x1080.jpg&pid=hp"
                    }
                >
                    <WeatherAccordeon />
                </HeroImage>
            </ErrorBoundary>
        </Suspense>
    );
};

export default App;
