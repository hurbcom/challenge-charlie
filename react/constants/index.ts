export const POSITIONS_TO_OPACITIES: Record<number, string> = {
    0: "--max-bg-opacity",
    1: "--medium-bg-opacity",
    2: "--min-bg-opacity",
};

export const SIZE = {
    mobileS: "320px",
    mobileM: "375px",
    mobileL: "425px",
    tablet: "768px",
    laptop: "1024px",
    laptopL: "1440px",
    desktop: "2560px",
};

export const DEVICES = {
    mobileS: `(min-width: ${SIZE.mobileS})`,
    mobileM: `(min-width: ${SIZE.mobileM})`,
    mobileL: `(min-width: ${SIZE.mobileL})`,
    tablet: `(min-width: ${SIZE.tablet})`,
    laptop: `(min-width: ${SIZE.laptop})`,
    laptopL: `(min-width: ${SIZE.laptopL})`,
    desktop: `(min-width: ${SIZE.desktop})`,
    desktopL: `(min-width: ${SIZE.desktop})`,
};
