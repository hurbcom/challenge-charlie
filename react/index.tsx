import React, { StrictMode } from "react";
import { hydrateRoot } from "react-dom/client";
import App from "./App";

const root = document.getElementById("react-app");

if (root) {
    hydrateRoot(
        root,
        <StrictMode>
            <App />
        </StrictMode>
    );
}
