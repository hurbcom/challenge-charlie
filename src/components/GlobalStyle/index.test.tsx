import { render } from "@testing-library/react";
import GlobalStylesComposed from ".";

describe("<GlobalStyles />", () => {
    test("renders global css", () => {
        render(<GlobalStylesComposed />);
        expect(document.head).toMatchSnapshot();
    });
});
