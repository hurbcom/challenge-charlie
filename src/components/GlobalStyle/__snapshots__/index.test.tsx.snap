// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalStyles /> renders global css 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.3.9"
  />
</head>
`;
